src/cpu/o3/lsq_unit.cc

```diff
@@ -1404,16 +1404,20 @@ bool LSQUnit::insertStoreBuffer(Addr vaddr, Addr paddr, uint8_t* datas, uint64_t
 void
 LSQUnit::storeBufferEvictToCache()
 {
-    if (isStoreBlocked) {
-        return;
-    }
-    if (storeBuffer.size() == 0) {
+    if (storeBufferFlushing && storeBuffer.size() == 0) [[unlikely]] {
         assert(storeBuffer.unsentSize() == 0);
         storeBufferFlushing = false;
         cpu->activityThisCycle();
+    }
+
+    // write request will stall one cycle
+    // so 2 cycle send one write request
+    if (lsq->getDcacheWriteStall()) {
+        lsq->setDcacheWriteStall(false);
         return;
```

Comment:
The Dcache requeset of this store may write DataSram after N cycles (4 if Hit, 10+/100+ if miss) arriving at Dcache, (I feel that the blocking should be done at that time instead of blocking directly in the next cycle?